libfreerdp/codec/region.c

```diff
@@ -132,7 +132,10 @@ static RECTANGLE_16 *region16_extents_noconst(REGION16 *region)
 
 BOOL rectangle_is_empty(const RECTANGLE_16 *rect)
 {
-	return (rect->left + rect->top + rect->right + rect->bottom) ? TRUE : FALSE;
+    /* An rectangle with width <= 0 or height <= 0 should be regarded
+     * as empty.
+     */
```

Comment:
I really don't like the idea that an invalid rectangle with right > left or bottom > top would be considered as empty.